File: src/python/example2-1.py
'''
example2-1.py
Create an XML document including different nodes.
'''

from lxml import etree
from sys import stdout


docname = '../../data/out2-1py.xml'

root = etree.Element('root')
child1 = etree.SubElement(root, 'child1')
child2 = etree.SubElement(root, 'child2', foo='foo', bar='bar')
child3 = etree.SubElement(child2, 'child3')
child3.text = 'some text'

tree = etree.ElementTree(root)
tree.write(stdout, pretty_print=True, xml_declaration=True, encoding='utf-8')
tree.write(docname, pretty_print=True, xml_declaration=True, encoding='utf-8')

File: src/c/example2-1.c
/*
example2-1.c
Create an XML document including different nodes.
*/

#include <stdio.h>

#include <libxml/parser.h>
#include <libxml/tree.h>



int
main(int argc, char **argv)
{
    xmlDocPtr doc = NULL;
    xmlNodePtr root = NULL, child1 = NULL, child2 = NULL, child3 = NULL;
    char *docname = "../../data/out2-1c.xml";


    doc = xmlNewDoc(BAD_CAST "1.0");
    root = xmlNewNode(NULL, BAD_CAST "root");
    xmlDocSetRootElement(doc, root);

    child1 = xmlNewChild(root, NULL, BAD_CAST "child1", NULL);

    child2 = xmlNewChild(root, NULL, BAD_CAST "child2", NULL);
    xmlNewProp(child1, BAD_CAST "foo", BAD_CAST "foo");
    xmlNewProp(child1, BAD_CAST "bar", BAD_CAST "bar");
    
    child3 = xmlNewChild(child2, NULL, BAD_CAST "child3", NULL);
    xmlNodeAddContent(child3, BAD_CAST "some text");

    xmlSaveFormatFileEnc("-", doc, "UTF-8", 1);
    xmlSaveFormatFileEnc(docname, doc, "UTF-8", 1);

    xmlFreeDoc(doc);

    return(0);
}





















